code/modules/antagonists/heretic/knowledge/moon_lore.dm

```diff
@@ -0,0 +1,266 @@
+		/datum/heretic_knowledge/summon/ashy,
+	)
+	spell_to_add = /datum/action/cooldown/spell/aoe/moon_ringleader
+	cost = 1
+	route = PATH_MOON
+
+/datum/heretic_knowledge/ultimate/moon_final
+	name = "The Last Act"
+	desc = "The ascension ritual of the Path of Moon. \
+		Bring 3 corpses with more than 50 brain damage to a transmutation rune to complete the ritual. \
+		When completed, you become a harbinger of madness, all your spells will get a lower cooldown \
+		and become stronger. You will also passivly release hallucinations, decrease sanity, and increase \
+		confusion. If their sanity is low they will start taking brain damage and go blind."
+	gain_text = "We dived down towards the crowd, his soul splitting off in search of greater venture \
+		for where the Ringleader had started the parade, I shall continue it unto the suns demise \
+		WITNESS MY ASCENSION, THE MOON SMILES ONCE MORE AND FOREVER MORE IT SHALL!"
+	adds_sidepath_points = 1
+	route = PATH_MOON
+
+/datum/heretic_knowledge/ultimate/moon_final/is_valid_sacrifice(mob/living/sacrifice)
+
+	var/brain_damage = sacrifice.get_organ_loss(ORGAN_SLOT_BRAIN)
+	// Checks if our target has enough brain damage
+	if(brain_damage<50)
+		return FALSE
+
+	return ..()
+
+/datum/heretic_knowledge/ultimate/moon_final/on_finished_recipe(mob/living/user, list/selected_atoms, turf/loc)
+	. = ..()
+	priority_announce("[generate_heretic_text()] Laugh, for the ringleader [user.real_name] has ascended! The truth shall finally devour the lie! [generate_heretic_text()]","[generate_heretic_text()]", ANNOUNCER_SPANOMALIES)
+
+	user.client?.give_award(/datum/award/achievement/misc/moon_ascension, user)
+	user.add_traits(TRAIT_MADNESS_IMMUNE)
+
+	RegisterSignal(user, COMSIG_LIVING_LIFE, PROC_REF(on_life))
+
+	var/datum/action/cooldown/spell/pointed/moon_smile/smile = locate() in user.actions
+	if(smile)
+		smile.cooldown_time *= 0.66 // Lower cooldown
+		smile.cast_range +=6 // Longer cast range
+
+	var/datum/action/cooldown/spell/pointed/projectile/moon_parade/lunar_parade = locate() in user.actions
+	if(lunar_parade)
+		lunar_parade.cooldown_time *= 0.66 // Lower cooldown
+
+	var/datum/action/cooldown/spell/aoe/moon_ringleader/ringleader_rise = locate() in user.actions
+	if(ringleader_rise)
+		ringleader_rise.cooldown_time *= 0.66 // Lower cooldown
+		ringleader_rise.aoe_radius +=3 // Bigger AoE
+
+/datum/heretic_knowledge/ultimate/moon_final/proc/on_life(mob/living/source, seconds_per_tick, times_fired)
+	SIGNAL_HANDLER
+
+	for(var/mob/living/carbon/carbon_view in view(7, source))
+		var/carbon_sanity=carbon_view.mob_mood.sanity
+
+		visible_hallucination_pulse(
+		center = get_turf(source),
+		radius = 7,
+		hallucination_duration = 20 SECONDS)
+
+		if(IS_HERETIC_OR_MONSTER(carbon_view))
+			continue
+		carbon_view.adjust_confusion(2 SECONDS)
+		carbon_view.mob_mood.set_sanity(carbon_sanity-5)
+		if(carbon_sanity<30)
+			carbon_view.balloon_alert(carbon_view, "you feel your mind begining to rend!")
```

Comment:
Really loong balloon alert, maybe needs to be shortened

Reply:
Turned it into a to_chat instead